### ISSUE TITLE
hover run error

### ISSUE THREAD

**Issue by windzhu** [go-flutter-desktop/go-flutter#420]
hover run
hover: Building flutter bundle
hover: Using engine from cache
hover: Failed to copy FlutterEmbedder.framework: symlink Versions/Current/FlutterEmbedder /Users/jiang/MyWorks/flutter/go-flutter-desktop/examples/pointer_demo/go/build/outputs/darwin/FlutterEmbedder.framework/FlutterEmbedder: file exists

**Comment by jld3103** [comment]
Is there no log line about cleaning the output/build directory? If not you must be running using the `--skip-flutter-build-bundle` flag, right?

**Comment by pchampio** [comment]
@jld3103 there is an inconsistency between `hover run` and `hover build`:
```
$ hover run --help
      --omit-embedder             Don't (re)compile 'go-flutter' source code, useful when only working with Dart code
      --omit-flutter              Don't (re)compile the current Flutter project, useful when only working with Golang code (plugin) 
```
```
$ hover build --help
      --skip-engine-download        Skip donwloading the Flutter Engine and artifacts.
      --skip-flutter-build-bundle   Skip the 'flutter build bundle' step.
```

**Comment by jld3103** [comment]
I know, but the build flags are only meant for docker really. Maybe we should hide them.

**Comment by pchampio** [comment]
@windzhu it would really help us if you provide the ouput of `hover doctor` (as indicated by the github issue template)!

Does this issue happens on every run?

**Comment by windzhu** [comment]
hover doctor
hover: Hover version v0.41.2 running on darwin
hover: Sharing flutter version
Flutter 1.18.0-9.0.pre.93 • channel master •
https://github.com/flutter/flutter.git
Framework • revision 06f81324d3 (2 days ago) • 2020-05-02 06:09:01 -0400
Engine • revision d0bcc6980e
Tools • Dart 2.9.0 (build 2.9.0-5.0.dev eca696c026)
hover: Flutter engine commit: https://github.com/flutter/engine/commit/d0bcc6980e4eb70eb94eced8b38fdffc2aa93443
hover: ⚠ The go-flutter project tries to stay compatible with the beta channel of Flutter.
hover: ⚠     It's advised to use the beta channel: `flutter channel beta`
hover: Finding out the C compiler version
Apple clang version 11.0.3 (clang-1103.0.32.59)
Target: x86_64-apple-darwin19.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
hover: Sharing the content of go.mod
module github.com/go-flutter-desktop/examples/stocks/desktop

go 1.12

require (
	github.com/go-flutter-desktop/go-flutter v0.40.0
	github.com/pkg/errors v0.9.1
)
hover: Warning: No hover.yaml file found: open go/hover.yaml: no such file or directory
hover: Sharing the content of go/cmd
go/cmd/main.go	go/cmd/options.go

**Comment by windzhu** [comment]
go version go1.14.2 darwin/amd64

**Comment by windzhu** [comment]
The first can run successfully, and then an error occurs every time.
cmd hover run --omit-embedder  can run successfully.
cmd hover run / build,   I need to delete exists directory,  rm -rf /Users/jiang/MyWorks/flutter/go-flutter-desktop/examples/pointer_demo/go/build/outputs/darwin/FlutterEmbedder.framework

**Comment by pchampio** [comment]
Maybe an anti virus is locking the `go/build/outputs/darwin/FlutterEmbedder.framework` file.
This would explain why ` hover run / build` cannot properly clean the `go/build/outputs/darwin/` before the copy happen???? :no_good:

Can you please provide the output of the failing `hover run`?

**Comment by ohsory1324** [comment]
In my case, `hover build darwin` works very well, but `hover run` is not working. I'm temporarily using `rm -rf go/build && hover run` command and it works. This is my `hover doctor`.

```
hover: Hover version v0.41.3 running on darwin
hover: Sharing flutter version
Flutter 1.15.17 • channel beta • https://github.com/flutter/flutter.git
Framework • revision 2294d75bfa (9 weeks ago) • 2020-03-07 00:28:38 +0900
Engine • revision 5aff311948
Tools • Dart 2.8.0 (build 2.8.0-dev.12.0 9983424a3c)
hover: Flutter engine commit: https://github.com/flutter/engine/commit/5aff3119480996ca014ec0f8d26d74db617b5852
hover: Finding out the C compiler version
Apple clang version 11.0.3 (clang-1103.0.32.59)
Target: x86_64-apple-darwin19.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
hover: Sharing the content of go.mod
module spotuprofessional/go

go 1.14

require (
	github.com/go-flutter-desktop/go-flutter v0.40.1
	github.com/go-flutter-desktop/plugins/path_provider v0.4.0
	github.com/go-flutter-desktop/plugins/shared_preferences v0.4.3
	github.com/go-flutter-desktop/plugins/video_player v0.0.3
	github.com/pkg/errors v0.9.1
)
hover: Sharing the content of hover.yaml
application-name: ""
executable-name: ""
package-name: ""
license: ""
target: lib/main_desktop.dart
branch: ""
cache-path: ""
opengl: ""
engine-version: ""
hover: Sharing the content of go/cmd
go/cmd/import-path_provider-plugin.go	go/cmd/import-shared_preferences-plugin.go	go/cmd/import-video_player-plugin.go	go/cmd/main.go	go/cmd/options.go
```

**Comment by CaiJingLong** [comment]
@ohsory1324 
I also use the command to run application with hover.

**Comment by marquesinijatinha** [comment]
having the same issue on `hover run`:
```
hover: Building flutter bundle
hover: Using engine from cache
hover: Failed to copy FlutterEmbedder.framework: symlink Versions/Current/FlutterEmbedder /Users/marquesini/workspace/littlelight-go/go/build/outputs/darwin/FlutterEmbedder.framework/FlutterEmbedder: file exists
```

```hover build darwin``` runs fine tho:
```
hover: Cleaning the build directory
hover: Building flutter bundle
hover: Using engine from cache
hover: Compiling 'go-flutter' and plugin

```

**Comment by jld3103** [comment]
Fixed in hover `v0.41.6`. Update and then it should work.
